Setting CAN_STACK_DISABLE_THREADS on causes compiler error

**Describe the bug**
When I try to build the project with the option `CAN_STACK_DISABLE_THREADS,` it fails to compile in my environment. 

```
/home/alfonso.ros/gh/AgIsoStack-plus-plus/isobus/include/isobus/isobus/can_control_function.hpp:74:22: error: ‘atomic_uint8_t’ in namespace ‘std’ does not name a type
   74 |                 std::atomic_uint8_t address; ///< The address of the control function
```

Seems like `atomic_uint8_t` is optionally defined depending on the platform and the solution should be simply about using `std::atomic<uint8_t>` instead:

```diff
diff --git a/isobus/include/isobus/isobus/can_control_function.hpp b/isobus/include/isobus/isobus/can_control_function.hpp
index db867c5..81c9997 100644
--- a/isobus/include/isobus/isobus/can_control_function.hpp
+++ b/isobus/include/isobus/isobus/can_control_function.hpp
@@ -14,6 +14,7 @@
 #include "isobus/isobus/can_NAME.hpp"
 #include "isobus/utility/thread_synchronization.hpp"
 
+#include <atomic>
 #include <memory>
 #include <string>
 
@@ -71,7 +72,7 @@ namespace isobus
        const Type controlFunctionType; ///< The Type of the control function
        NAME controlFunctionNAME; ///< The NAME of the control function
        bool claimedAddressSinceLastAddressClaimRequest = false; ///< Used to mark CFs as stale if they don't claim within a certain time
-       std::atomic_uint8_t address; ///< The address of the control function
+       std::atomic<uint8_t> address; ///< The address of the control function
        const std::uint8_t canPortIndex; ///< The CAN channel index of the control function
    };
```

** Steps to reproduce:

- `cmake -S. -Bbuild -DCAN_STACK_DISABLE_THREADS=ON`
- `cmake --build build`

**Environment (please complete the following information):**

- OS: Ubuntu 22.04.5 LTS
- Compiler: c++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
- CAN Driver: socketcan

--- a/isobus/include/isobus/isobus/can_control_function.hpp
+++ b/isobus/include/isobus/isobus/can_control_function.hpp
@@ -14,6 +14,7 @@
 #include "isobus/isobus/can_NAME.hpp"
 #include "isobus/utility/thread_synchronization.hpp"
 
+#include <atomic>
 #include <memory>
 #include <string>
 
@@ -71,7 +72,7 @@ namespace isobus
 		const Type controlFunctionType; ///< The Type of the control function
 		NAME controlFunctionNAME; ///< The NAME of the control function
 		bool claimedAddressSinceLastAddressClaimRequest = false; ///< Used to mark CFs as stale if they don't claim within a certain time
-		std::atomic_uint8_t address; ///< The address of the control function
+		std::atomic<std::uint8_t> address; ///< The address of the control function
 		const std::uint8_t canPortIndex; ///< The CAN channel index of the control function
 	};
 

